bin/dartdoc.dart

```diff
@@ -249,104 +255,108 @@ main(List<String> arguments) async {
-  parser.addMultiOption('category-order',
-      help:
-          'A list of package names to place first when grouping libraries in packages. '
-          'Unmentioned categories are sorted after these. (deprecated, replaced by package-order)',
-      splitCommas: true);
+  parser.addOption('input', help: 'Path to source directory.');
+  parser.addFlag('json',
+      help: 'Prints out progress JSON maps. One entry per line.',
+      defaultsTo: false,
+      negatable: true);
+  parser.addOption('output',
+      help: 'Path to output directory.', defaultsTo: defaultOutDir);
   parser.addMultiOption('package-order',
       help:
           'A list of package names to place first when grouping libraries in packages. '
           'Unmentioned categories are sorted after these.',
       splitCommas: true);
-  parser.addFlag('auto-include-dependencies',
-      help:
-          'Include all the used libraries into the docs, even the ones not in the current package or "include-external"',
-      negatable: false,
-      defaultsTo: false);
   parser.addFlag('pretty-index-json',
       help:
           "Generates `index.json` with indentation and newlines. The file is larger, but it's also easier to diff.",
       negatable: false,
       defaultsTo: false);
-  parser.addOption('ambiguous-reexport-scorer-min-confidence',
-      help:
-          'Minimum scorer confidence to suppress warning on ambiguous reexport.',
-      defaultsTo: "0.1",
-      hide: true);
-  parser.addFlag('verbose-warnings',
-      help: 'Display extra debugging information and help with warnings.',
-      negatable: true,
-      defaultsTo: true);
-  parser.addFlag('hide-sdk-text',
+  parser.addOption('rel-canonical-prefix',
+      help: 'If provided, add a rel="canonical" prefixed with provided value. '
+          'Consider using if\nbuilding many versions of the docs for public '
+          'SEO; learn more at https://goo.gl/gktN6F.');
+  parser.addFlag('sdk-docs',
+      help: 'Generate ONLY the docs for the Dart SDK.', negatable: false);
+  parser.addOption('sdk-readme',
+      help: 'Path to the SDK description file.  Deprecated (ignored)');
+  parser.addOption('sdk-dir',
+      help: 'Path to the SDK directory',
+      defaultsTo: defaultSdkDir.absolute.path);
+  parser.addFlag('show-warnings',
+      help: 'Display warnings.', negatable: false, defaultsTo: false);
+  parser.addFlag('show-progress',
+      help: 'Display progress indications to console stdout', negatable: false);
+  parser.addFlag('use-categories',
       help:
-          "Drop all text for SDK components.  Helpful for integration tests for dartdoc, probably not useful for anything else.",
-      negatable: true,
-      defaultsTo: false,
-      hide: true);
-  parser.addFlag('json',
-      help: 'Prints out progress JSON maps. One entry per line.',
-      defaultsTo: false,
-      negatable: true);
+          'Group libraries from the same package in the libraries sidebar. (deprecated, ignored)',
```

Comment:
=> (Deprecated, ignored) 

?

Reply:
use-categories was marked deprecated, ignored back in [#1641](https://github.com/dart-lang/dartdoc/pull/1641/files#diff-32a35d5cbe31636791d779e9c772144bR349) because the meaning of "category" changed and we no longer require this flag to turn on what was category mode.  Dartdoc now knows whether to do that by context and we just ignore this flag if it gets passed in.

The diff looks weird because I sorted the options here to better keep track of them visually, as I'm going to be doing major surgery on them soon.